daemon/lokinet-vpn.cpp

```diff
@@ -98,119 +104,102 @@ namespace
+  cli.add_option(
+         "--log-level", options.logLevel, "Log verbosity level, see log levels for accepted values")
+      ->type_name("LEVEL")
+      ->capture_default_str();
 
-  // clang-format off
-  opts.add_options()
-    ("v,verbose", "Verbose", cxxopts::value<bool>())
-    ("h,help", "help", cxxopts::value<bool>())
-    ("kill", "kill the daemon", cxxopts::value<bool>())
-    ("up", "put vpn up", cxxopts::value<bool>())
-    ("down", "put vpn down", cxxopts::value<bool>())
-    ("exit", "specify exit node address", cxxopts::value<std::string>())
-    ("rpc", "rpc url for lokinet", cxxopts::value<std::string>())
-    ("endpoint", "endpoint to use", cxxopts::value<std::string>())
-    ("token", "exit auth token to use", cxxopts::value<std::string>())
-    ("auth", "exit auth token to use", cxxopts::value<std::string>())
-    ("status", "print status and exit", cxxopts::value<bool>())
-    ("range", "ip range to map", cxxopts::value<std::string>())
-    ;
-  // clang-format on
-  oxenmq::address rpcURL("tcp://127.0.0.1:1190");
-  std::string exitAddress;
-  std::string endpoint = "default";
-  std::string token;
-  std::optional<std::string> range;
-  oxenmq::LogLevel logLevel = oxenmq::LogLevel::warn;
-  bool goUp = false;
-  bool goDown = false;
-  bool printStatus = false;
-  bool killDaemon = false;
   try
   {
-    const auto result = opts.parse(argc, argv);
-
-    if (result.count("help") > 0)
-    {
-      std::cout << opts.help() << std::endl;
-      return 0;
-    }
-
-    if (result.count("verbose") > 0)
-    {
-      logLevel = oxenmq::LogLevel::debug;
-    }
-    goUp = result.count("up") > 0;
-    goDown = result.count("down") > 0;
-    printStatus = result.count("status") > 0;
-    killDaemon = result.count("kill") > 0;
-
-    extract_option(result, "rpc", rpcURL);
-    extract_option(result, "exit", exitAddress);
-    extract_option(result, "endpoint", endpoint);
-    extract_option(result, "token", token);
-    extract_option(result, "auth", token);
-    extract_option(result, "range", range);
+    cli.parse(argc, argv);
+  }
+  catch (const CLI::ParseError& e)
+  {
+    return cli.exit(e);
   }
-  catch (const cxxopts::option_not_exists_exception& ex)
+
+  try
   {
-    return exit_error(2, "{}\n{}", ex.what(), opts.help());
+    if (options.verbose)
+      options.logLevel = oxenmq::LogLevel::debug;
   }
-  catch (std::exception& ex)
+  catch (const CLI::OptionNotFound& e)
   {
-    return exit_error(2, "{}", ex.what());
+    cli.exit(e);
   }
+  catch (const CLI::Error& e)
+  {
+    cli.exit(e);
+  };
 
-  int num_commands = goUp + goDown + printStatus + killDaemon;
-
-  if (num_commands == 0)
-    return exit_error(3, "One of --up/--down/--status/--kill must be specified");
-  if (num_commands != 1)
-    return exit_error(3, "Only one of --up/--down/--status/--kill may be specified");
+  switch (options.vpnUp + options.vpnDown + options.printStatus + options.killDaemon)
```

Comment:
implicit conversion to `int` feels like foot cannon fodder

Reply:
Perhaps keep it explicit via the `int num_commands = ...`